applications/SwimmingDEMApplication/custom_elements/shell_rigid.cpp

```diff
@@ -1961,52 +1961,28 @@ int  ShellRigid::Check(const ProcessInfo& rCurrentProcessInfo)
 {
     KRATOS_TRY
 
-
-
     //verify that the variables are correctly initialized
-    if(VELOCITY.Key() == 0)
-        KRATOS_THROW_ERROR(std::invalid_argument,"VELOCITY has Key zero! (check if the application is correctly registered","");
-    if(DISPLACEMENT.Key() == 0)
-        KRATOS_THROW_ERROR(std::invalid_argument,"DISPLACEMENT has Key zero! (check if the application is correctly registered","");
-    if(ACCELERATION.Key() == 0)
-        KRATOS_THROW_ERROR(std::invalid_argument,"ACCELERATION has Key zero! (check if the application is correctly registered","");
-    if(DENSITY.Key() == 0)
-        KRATOS_THROW_ERROR(std::invalid_argument,"DENSITY has Key zero! (check if the application is correctly registered","");
-    if(BODY_FORCE.Key() == 0)
-        KRATOS_THROW_ERROR(std::invalid_argument,"BODY_FORCE has Key zero! (check if the application is correctly registered","");
-    if(THICKNESS.Key() == 0)
-        KRATOS_THROW_ERROR(std::invalid_argument,"THICKNESS has Key zero! (check if the application is correctly registered","");
+    KRATOS_ERROR_IF(VELOCITY.Key() == 0) << "VELOCITY Key is 0. Check if the application was correctly registered." << std::endl;
+    KRATOS_ERROR_IF(DISPLACEMENT.Key() == 0) << "DISPLACEMENT Key is 0. Check if the application was correctly registered." << std::endl;
+    KRATOS_ERROR_IF(ACCELERATION.Key() == 0) << "ACCELERATION Key is 0. Check if the application was correctly registered." << std::endl;
+    KRATOS_ERROR_IF(DENSITY.Key() == 0) << "DENSITY Key is 0. Check if the application was correctly registered." << std::endl;
+    KRATOS_ERROR_IF(BODY_FORCE.Key() == 0) << "BODY_FORCE Key is 0. Check if the application was correctly registered." << std::endl;
+    KRATOS_ERROR_IF(THICKNESS.Key() == 0) << "THICKNESS Key is 0. Check if the application was correctly registered." << std::endl;
```

Comment:
```suggestion
```

since variables keys are assigned at compile time for a long time already this can be removed

Reply:
You mean that the key can never be 0? 
Because probably I will need to remove it in many other places as well.

Reply:
@philbucher this is still pending. Should I remove all places where we check key = 0?

Reply:
according to @pooyan-dadvand yes this can be removed everywhere.

@pooyan-dadvand it would be nice if you could confirm

Reply:
I confirm!

Reply:
Thanks Pooyan! Then, we have a problem in the core, here:
https://github.com/KratosMultiphysics/Kratos/blob/1e5ac6627da940de987503108adb24540c4a2bb3/kratos/includes/checks.h#L172
I think we should open a new PR to fix that and everywhere where it is called.
What is the correct current way of checking if a variable is registered and usable?

Reply:
see #7725

Reply:
Ok, that's good, but in these case here I think that @GuillermoCasas still wants to make sure that the application is consistent with itself, so it checks that those variables are registered. Not 'registered correctly' but just registered. How can we do that? I macro would be nice.

Reply:
I think it would simply not compile if sth is wrong
But lets hear what pooyan has to say

Reply:
Being more specific, I am actually thinking of a macro like:
`KRATOS_CHECK_VARIABLE_IS_REGISTERED(VELOCITY)`
which would execute:
`KratosComponents<VariableData>::Has("VELOCITY");`

Reply:
@pooyan-dadvand